content/en/platform/corda/4.11/enterprise/ledger-recovery-flow.md

```diff
@@ -0,0 +1,166 @@
+
+A `SenderDistribution` record contains the following transaction metadata:
+
+* Transaction ID
+* Receiver peer ID (the hashed value of `CordaX500Name`)
+* Receiver `StatesToRecord` value
+
+{{< note >}}
+The `SendTransactionFlow` has been enhanced to infer the value for receiver `StatesToRecord` based on the type of sessions passed into its constructor:
+
+* val participantSessions: Set<FlowSession> => default to ONLY_RELEVANT
+* val observerSessions: Set<FlowSession> => default to ALL_VISIBLE
+{{< /note >}}
+
+Upon storing the `SenderDistribution` records for a transaction, the sender node also generates a single `ReceiverDistribution` record.
+
+A `ReceiverDistribution` record contains the following transaction metadata:
+
+* Transaction ID
+* Sender peer ID (the hashed value of `CordaX500Name`)
+* A distribution list comprising two sections:
+  * An AES-encrypted map of all peers and their associated `StatesToRecord` value.
+  * A tamper-proof cleartext `senderRecordedTimestamp`, indicating when the sender records were generated. The same timestamp will be used upon storing the `ReceiverDistribution` record at each of the receiving node peers. This is required to enable synchronized record comparisons across peers when performing transaction recovery.
+
+The generated `ReceiverDistribution` record is then shared with all other peer sessions (both participants and observers) to the transaction, which stores it in their local `receiver_distribution_records` database table. 
+
+{{< note >}}
+A receiver cannot decrypt the actual contents of the distribution list within that record, but shares it back to a transaction initiating node upon recovery.
+{{< /note >}}
+
+Both sender and receiver distribution records use the same composite key type for uniquely storing records. The `PersistentKey` contains the following fields:
+
+* Incremental sequence Number (Long)
+* Timestamp (Instant)
+* Timestamp discriminator (Int)
+
+The timestamp discriminator allows for storing of records generated at the same time but within different transactions.
+
+The sender and receiver distribution records are subsequently used to enable Ledger Recovery.
+
+## Ledger Recovery flow
+
+The Ledger Recovery flow takes the following parameters:
+
+```
+private val recoveryPeers: Collection<Party>,
+private val timeWindow: RecoveryTimeWindow,
+private val useAllNetworkNodes: Boolean = false,
+private val transactionRole: TransactionRole = TransactionRole.ALL,
+private val dryRun: Boolean = false,
+private val optimisticInitiatorRecovery: Boolean = false,
+```
+
+These parameters are described in the following section.
+
+
+### Ledger Recovery flow parameters
+
+#### `recoveryPeers`
+
+`recoveryPeers` refers to a list of peer nodes to use for recovery.  This parameter is mandatory.
+
+#### `timeWindows`
+
+`timeWindow` refers to the recovery time window and defines a `fromTime` and `untilTime`. This parameter is mandatory. 

```

Comment:
`timeWindow` refers to the recovery time window and defines a `fromTime` and `untilTime`.
If a value is not specified by the user, the flow will attempt to use the Corda Network or Node Configuration configured time window.